src/main/java/seedu/address/logic/parser/MarkTaskCommandParser.java

```diff
@@ -33,17 +33,12 @@ public MarkTaskCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkTaskCommand.MESSAGE_USAGE));
         }
 
-        try {
-            // get project and task indexes
-            Index projectIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_PROJECT).get());
-            List<Index> taskIndexes = ParserUtil.parseIndexes(argMultimap.getValue(PREFIX_TASK).get());
-            assert taskIndexes.size() > 0;
-
-            return new MarkTaskCommand(projectIndex, taskIndexes);
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkTaskCommand.MESSAGE_USAGE), pe);
-        }
+        // get project and task indexes
+        Index projectIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_PROJECT).get());
+        List<Index> taskIndexes = ParserUtil.parseIndexes(argMultimap.getValue(PREFIX_TASK).get());
+        assert taskIndexes.size() > 0;
+
+        return new MarkTaskCommand(projectIndex, taskIndexes);
```

Comment:
I believe this solves the "duplicate index error message not showing up" issue right?

Reply:
It's not this part, it's the removal of the try-catch block from the `parse` function (this block unintentionally caught some exceptions).